src/reference.ts

```diff
@@ -34,41 +33,42 @@ export function getDataAttributeOptions(reference) {
 /**
  * Polyfills the virtual reference (plain object) with Element.prototype props
  * Mutating because DOM elements are mutated, adds `_tippy` property
- * @param {Object} virtualReference
  */
-export function polyfillElementPrototypeProperties(virtualReference) {
-  const polyfills = {
+export function polyfillElementPrototypeProperties(
+  virtualReference: VirtualReference,
```

Comment:
```suggestion
  virtualReference: VirtualReference & Record<string, any>,
```

I don't know if this is the best way, but it will help with `;(virtualReference as any)[key] = polyfills[key]`. Also `const polyfills: Record<string, any>`